From fcf3dd354f9032336cac79314a6086226adf02cf Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Sun, 9 Jun 2013 20:47:06 -0400
Subject: [PATCH] Make 01-boot-stack support requirements.txt.

Updates install.d/01-boot-stack so that it looks in requirements.txt
as well as the older (legacy) tools/pip-requires files when
attempting to install dependencies for our client libraries.

This fixes a rather cryptic issue that crept in this morning
with quantumclient due to the fact that they just moved over
to use requirements.txt... this would essentially cause some
of the quantumclient dependencies to get incorrectly easy_install'ed and
thus have bad permissions (on some of the egg-info files).

Change-Id: I488be161c45408727adeceb26f1f5c6b1de01430
---
 elements/boot-stack/install.d/01-boot-stack | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/elements/boot-stack/install.d/01-boot-stack b/elements/boot-stack/install.d/01-boot-stack
index b5b6717..c365a87 100755
--- a/elements/boot-stack/install.d/01-boot-stack
+++ b/elements/boot-stack/install.d/01-boot-stack
@@ -46,7 +46,10 @@ for client in nova quantum glance heat; do
   pip install cliff
 
   pushd /opt/stack/$repo
-  if [ -e tools/pip-requires ]; then
+  if [ -e requirements.txt ]; then
+    PIP_INDEX_URL=http://pypi.openstack.org/openstack/ \
+      pip install -r requirements.txt
+  elif [ -e tools/pip-requires ]; then
     PIP_INDEX_URL=http://pypi.openstack.org/openstack/ \
       pip install -r tools/pip-requires
   fi
-- 
1.8.1.4

